packages/astro/src/runtime.ts

```diff
@@ -120,6 +129,8 @@ async function load(config: RuntimeConfig, rawPathname: string | undefined): Pro
       }
 
       let data: any[] = await loadData({ params: currentParams });
+      if (!data) throw new Error(`[createCollection] \`data()\` returned nothing (empty data)"`);
+      if (!Array.isArray(data)) data = [data]; // note: this is supposed to be a little friendlier to the user, but should we error out instead?
```

Comment:
Another UX improvement: if the user didn’t return an array of data, convert it into one for them. While this may cause some weirdness, ideally this makes the users’ own errors clearer (and IMO this is better than erring out if the user didn’t return an array)